openmsupply/open-msupply
# 1443 repack - UI

Fixes #1443 

# 👩🏻‍💻 What does this PR do? 
Changed how the UI for repacks would look from the original mock since it could be confusing for users. In the mock UI, users could potentially edit the Stock Line and Repack at the same time, or even if this behaviour is restricted to only saving Edit Stock Line or Repack, users may believe they can do both at the same time... So have refactored a bit and shuffled the `View Stock` UI.

Allows a user to select a `Stock Line` by clicking on a checkbox, and refactored the `Edit Stock Line Modal` to open once a user has selected a row and clicked on the new `Edit Stock Line` button at the top right of the view. 

Adds a `Repack` button which will also open when a `Stock Line` has been selected and the button is pressed. 
![Screenshot 2023-05-31 at 9 05 48 AM](https://github.com/openmsupply/open-msupply/assets/61820074/24f16daa-551f-4a13-8a13-9973ef6a3908)
If an item has never been repacked, then nothing will show on the right of the screen, but if it has been repacked then a clickable list will show displaying these repacks as shown above.

Information about the repack can be shown on the right if a line has been clicked. 
![Screenshot 2023-05-31 at 10 01 05 AM](https://github.com/openmsupply/open-msupply/assets/61820074/c4760397-08e0-40b2-ab94-a270a8f6922b)
Or alternatively, the stock line can be repacked by clicking `New`.

# 🧪 How has/should this change been tested? 
- Go to `View Stock` 
- Click on a stock line checkbox 
- Click the `Repack` button
- Click new to split the stock line
- Enter `New Pack Size` and `New # of Packs`
- Click ok.
- Go back to stock line
- Click on the repack line
- See repacked details

## 💌 Any notes for the reviewer?
- #1867 Moving better error handling messages and printing to another issue since this was getting a bit big... 😅 

## 📃 Documentation

- [ ] New screenshots in `View Stock` docs
- [ ] Edit `View Stock - Editing a stock line` to show users how to open the modal.
- [ ] Add `Repack` to `View Stock` docs
